### ISSUE TITLE
Felaktig anmärkning på versionsnummer i namespace

### ISSUE THREAD

**Issue by nordvall** [skltp/vt-gen#1]
VirtualiseringGenerator.groovy kontrollerar formatet på namespacen i alla xsd-filer (utom _ext.xsd) under schemas/interactions i den aktuella domänen. Jag antar att det är viktigt pga att kontraktets namespace ligger till grund för routing-informationen i TAK.

Men scriptet ser ut att göra felaktig bedömning när det gäller kontrakt som är utökade, och därför innehåller en _ext.xsd-fil som refereras från den ordinarie xsd-filen. 

Här är ett exempel från domänen crm:scheduling, kontraktet GetAvailableTimeslots. Där ser schema-elementet ut så här:

`<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:core="urn:riv:crm:scheduling:1" xmlns:tns="urn:riv:crm:scheduling:GetAvailableDatesResponder:1" xmlns:m1="urn:riv:crm:scheduling:GetAvailableDatesResponder:1.1" targetNamespace="urn:riv:crm:scheduling:GetAvailableDatesResponder:1" elementFormDefault="qualified" attributeFormDefault="unqualified" version="1.1">
`

De regler som styr utformningen av namespace finns i RIV TA Tjänsteschema http://rivta.se/documents/ARK_0005/, främst regel #3). Ovanstående xml är såvitt jag kan se helt enligt exemplet i bilaga 2 ("icke-bakåtkompatibel utökning").

Kör man generatorn på denna domän får man följande fel:

`File to process: GetAvailableDatesInteraction_1.1_RIVTABP21.wsdl
File to process: GetAvailableDatesResponder_1.1.xsd
File to process: GetAvailableDatesResponder_1.1_ext.xsd
Incorrect version number for the following service contract! /path/to/riv.crm.scheduling/schemas/interactions/GetAvailableDatesInteraction. Version is: 1.1`

Felet verkar ligga i funktionen getServiceContractNameSpaceVersion, som undersöker **alla** refererade namespace som innehåller "urn:riv" och kontrollerar att de inte innehåller någon punkt i den sista delen. Det korrekta borde vara att helt enkelt undersöka innehållet i targetNamespace-attributet. 

mvh 
Mattias Nordvall, Inera

**Comment by EricIO** [comment]
Hej Mattias!

Tack för buggrapportering. Vi har också sett detta och jag tror att vi har löst det i _develop_ branchen. Vi har ännu inte gjort en release av denna branch ännu. Jag föreslår att du använder _develop_ tills vi har gjort en release (vilket vi borde hinna med idag).

**Comment by nordvall** [comment]
Ah, i develop fungerar det! 

Jag har dock ett förbättringsförslag angående funktionen getServiceContractNameSpaceVersion. Just nu plockar den ut alla deklarerade namespaces som innehåller strängen 'urn:riv' och försöker därefter lista ut vilket som är det rätta. 
Ni skulle få samma resultat genom att helt enkelt titta i attributet targetNamespace. Då skulle ni slippa each-loopen, if-satsen och dessutom inte vara beroende av att korrekt namespace kommer före övriga riv-namespace.

Jag ramlade in på detta pga att jag arbetar med releaser av tjänstekontrakt för Ineras räkning och gjorde några stickprov med generatorn för att se att den accepterar våra kontrakt. Det är alltså ingen brådska med någon ny release för min del.

**Comment by HansWiklund** [comment]
Ja, ditt förslag är en förbättring. Jag lägger upp en Jira om det.

Mvh Hans
